libensemble/tests/regression_tests/test_persistent_tasmanian_async.py

```diff
@@ -0,0 +1,90 @@
+"""
+Tests the async-mode of the Tasmanian generator function.
+
+Execute via one of the following commands (e.g. 3 workers):
+   mpiexec -np 4 python3 test_persistent_tasmanian_async.py
+   python3 test_persistent_tasmanian_async.py --nworkers 3 --comms local
+   python3 test_persistent_tasmanian_async.py --nworkers 3 --comms tcp
+
+When running with the above commands, the number of concurrent evaluations of
+the objective function will be 2, as one of the three workers will be the
+persistent generator.
+"""
+
+# Do not change these lines - they are parsed by run-tests.sh
+# TESTSUITE_COMMS: local
+# TESTSUITE_NPROCS: 4
+# TESTSUITE_OS_SKIP: OSX
+# TESTSUITE_EXTRA: true
+
+import sys
+import itertools
+import numpy as np
+
+# Import libEnsemble items for this test
+from libensemble.libE import libE
+from libensemble.sim_funcs.six_hump_camel import six_hump_camel_func
+from libensemble.gen_funcs.persistent_tasmanian import get_sparse_grid_inputs
+from libensemble.tools import parse_args, save_libE_output, add_unique_random_streams
+from time import time
+
+# Define some grid initializers.
+def tasmanian_init_global(num_dims):
+    # Note: if Tasmanian has been compiled with OpenMP support (i.e., the usual way)
+    #       libEnsemble calls cannot be made after the `import Tasmanian` clause
+    #       there is a conflict between the OpenMP environment and Python threading
+    #       thus Tasmanian has to be imported inside the `tasmanian_init` method
+    import Tasmanian
+    grid = Tasmanian.makeGlobalGrid(num_dims, 1, 6, "iptotal", "clenshaw-curtis")
+    grid.setDomainTransform(np.array([[-5.0, 5.0], [-2.0, 2.0]]))
+    return grid
+
+def tasmanian_init_localp(num_dims):
+    import Tasmanian
+    grid = Tasmanian.makeLocalPolynomialGrid(num_dims, 1, 3)
+    grid.setDomainTransform(np.array([[-5.0, 5.0], [-2.0, 2.0]]))
+    return grid
+
+# Get node info.
+nworkers, is_manager, libE_specs, _ = parse_args()
+if nworkers < 2:
+    sys.exit("Cannot run with a persistent worker if only one worker -- aborting...")
+
+# Create an async simulator function (must return 'x' and 'f').
+def sim_f(H, persis_info, sim_specs, _):
+    batch = len(H['x'])
+    H0 = np.zeros(batch, dtype=sim_specs['out'])
+    H0['x'] = H['x']
+    for i, x in enumerate(H['x']):
+        H0['f'][i] = six_hump_camel_func(x)
+    return H0, persis_info
+
+# Set up test parameters.
+num_dims = 2
+user_specs_arr = []
+user_specs_arr.append({
+    'refinement' : 'anisotropic',
+    'tasmanian_init' : lambda : tasmanian_init_global(num_dims),
```

Comment:
We should probably remove `num_dims` from the inputs here, the actual function inits the grid but also sets domain transform, which can only work with 2 dimensions.

Remove it from the second call below too.

Reply:
Fixed in a new PR.